contrib/cmd/runkperf/commands/data/configmaps/configmap.go

```diff
@@ -0,0 +1,389 @@
+			return err
+		}
+
+		const (
+			minWidth = 1
+			tabWidth = 12
+			padding  = 3
+			padChar  = ' '
+			flags    = 0
+		)
+		tw := tabwriter.NewWriter(os.Stdout, minWidth, tabWidth, padding, padChar, flags)
+		fmt.Fprintln(tw, "NAME\tSIZE\tGROUP_SIZE\tTOTAL\t")
+
+		// Build the label selector
+		// If no args are provided, list all configmaps with the label app=kperf
+		// If args are provided, list all configmaps with the label app=kperf and cmName in (args)
+		var labelSelector string
+		if cliCtx.NArg() == 0 {
+			labelSelector = "app=kperf"
+
+		} else {
+			args := cliCtx.Args()
+			namesStr := strings.Join(args, ",")
+			labelSelector = fmt.Sprintf("app=kperf, cmName in (%s)", namesStr)
+		}
+		cmMap := make(map[string][]int)
+		err = listConfigmapsByName(cmMap, labelSelector, clientset, namespace)
+
+		if err != nil {
+			return err
+		}
+
+		for key, value := range cmMap {
+			fmt.Fprintf(tw, "%s\t%d\t%d\t%d\n",
+				key,
+				value[0],
+				value[2]-value[1],
+				value[2],
+			)
+		}
+		return tw.Flush()
+	},
+}
+
+func prepareNamespace(kubeCfgPath string, namespace string) error {
+	if namespace == "" {
+		return fmt.Errorf("namespace cannot be empty")
+	}
+
+	if namespace == "default" {
+		return nil
+	}
+
+	ctx := context.Background()
+	kr := contributils.NewKubectlRunner(kubeCfgPath, namespace)
+	err := kr.CreateNamespace(ctx, 5*time.Minute, namespace)
+	if err != nil {
+		return fmt.Errorf("failed to create namespace %s: %v", namespace, err)
+	}
+	return nil
+}
+
+func checkConfigmapParams(size int, groupSize int, total int) error {
+	if size <= 0 {
+		return fmt.Errorf("size must be greater than 0")
+	}
+	if groupSize <= 0 {
+		return fmt.Errorf("group-size must be greater than 0")
+	}
+	if total <= 0 {
+		return fmt.Errorf("total amount must be greater than 0")
+	}
+	if groupSize > total {
+		return fmt.Errorf("group-size must be less than or equal to total")
+	}
+	if total%groupSize != 0 {
+		return fmt.Errorf("total must be divisible by group-size")
+	}
+	return nil
+}
+
+func newClientsetWithRateLimiter(kubeCfgPath string, qps float32, burst int) (*kubernetes.Clientset, error) {
+	config, err := clientcmd.BuildConfigFromFlags("", kubeCfgPath)
+	if err != nil {
+		return nil, err
+	}
+
+	config.QPS = qps
+	config.RateLimiter = flowcontrol.NewTokenBucketRateLimiter(qps, burst)
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+	return clientset, nil
+}
+
+var letterRunes = []rune("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ")
+
+func RandStringRunes(n int) string {
+	b := make([]rune, n)
+	for i := range b {
+		b[i] = letterRunes[rand.Intn(len(letterRunes))]
+	}
+	return string(b)
+}
+
+func createConfigmaps(cmName string, size int, groupSize int, total int, clientset *kubernetes.Clientset, namespace string) {
+	// Generate configmaps in parallel with fixed group size
+	// and random data
+	for i := 0; i < total; i = i + groupSize {
+		ownerID := i
+		var wg sync.WaitGroup
```

Comment:
you can try to use this one `https://pkg.go.dev/golang.org/x/sync/errgroup`
`wg.Wait` will return first error and then we can just return that error to caller.